### PR TITLE
[Enhancement] add font size option per language on registration UI

### PR DESCRIPTION
**Pull request by KimJeongSun**
## What are the changes the user will see?

Spanish-language users will see a more polished and organized register menu screen.

## Why am I making these changes?

There was a request from @AsdarDevelops  to make these changes!

## What are the changes from a developer perspective?

I modified `registration-form-ui-handler.ts` to allow font settings for each language.

### Screenshots/Videos

- before
image

- after
image

## Checklist
- [x] **I'm using `beta` as my base branch**
- [x] There is no overlap with another PR?
- [x] The PR is self-contained and cannot be split into smaller PRs?
- [x] Have I provided a clear explanation of the changes?
- [ ] Have I considered writing automated tests for the issue?
- [ ] If I have text, did I make it translatable and add a key in the English locale file(s)?
- [x] Have I tested the changes (manually)?
    - [x] Are all unit tests still passing? (`npm run test`)
- [x] Are the changes visual?
  - [x] Have I provided screenshots/videos of the changes?

### CODE REVIEW

**Comment by KimJeongSun** [comment]
@flx-sta 

Actually, while working on the above changes, I also planned to organize the German version. However, the problem with the German version is that the button sizes need to be adjusted as well.

Below is a screenshot where I only adjusted the button labels’ size: ([related commit](https://github.com/pagefaultgames/pokerogue/pull/3849/commits/b6396771fdd70db1c3ee4ddeec98943e7d23d070))
Screenshot 2024-08-28 at 12 46 34 AM

As you can see, in my previous attempt, the button sizes didn't reduce properly.

If possible, adding this feature might make it more flexible and better. Additionally, you can consider refactoring my code by inheriting classes, if possible! (It was a bit too much for me)

Of course, if it’s not possible to work on this immediately, we can proceed as is and add further developments later.

**Comment by CodeTappert** [comment]
Please also for german. I think the text is too big there too

**Comment by KimJeongSun** [comment]
@flx-sta 🙏

**Review by flx-sta — DISMISSED** [review]

File: src/ui/registration-form-ui-handler.ts [hunk 1]
```diff
@@ -5,6 +5,20 @@ import { Mode } from "./ui";
...
 import i18next from "i18next";
 
+
+interface LanguageSetting {
+  inputFieldFontSize?: string,
+  warningMessageFontSize?: string,
+  errorMessageFontSize?: string,
+}
+
+const languageSettings: { [key: string]: LanguageSetting } = {
```
[the remark this answers is not in the document]
Comment (KimJeongSun) [reply src/ui/registration-form-ui-handler.ts:15]: Next time, we should consider gathering the relevant settings from each source file (this, starter UI etc.) and putting them into JSON keys in a PR.

**Comment by KimJeongSun** [comment]
@flx-sta
can you make new PR to beta ( or PR toward this PR ) for DE updates? ( including resize button size + button label font size )?

**Comment by flx-sta** [comment]
> @flx-sta can you make new PR to beta ( or PR toward this PR ) for DE updates? ( including resize button size + button label font size )?

that would be @CodeTappert

File: src/locales/es/menu.json [hunk 2]
```diff
@@ -14,14 +14,14 @@
...
-  "invalidRegisterUsername": "El usuario solo puede contener letras, números y guiones bajos",
+  "invalidRegisterUsername": "El usuario solo puede contener letras, números y guiones bajos.",
   "invalidLoginPassword": "La contraseña no es válida",
   "invalidRegisterPassword": "La contraseña debe tener 6 o más caracteres.",
   "usernameAlreadyUsed": "El usuario ya está en uso",
   "accountNonExistent": "El usuario no existe",
   "unmatchingPassword": "La contraseña no coincide",
   "passwordNotMatchingConfirmPassword": "Las contraseñas deben coincidir",
-  "confirmPassword": "Confirmar Contra.",
+  "confirmPassword": "Confirmar Contraseña",
```
Comment (AsdarDevelops) [inline src/locales/es/menu.json:24]:
```suggestion
  "confirmPassword": "Confirmar contraseña",
```

**Review by AsdarDevelops — APPROVED** [review]
Thank you so much

File: src/ui/registration-form-ui-handler.ts [hunk 3]
```diff
@@ -50,7 +64,17 @@ export default class RegistrationFormUiHandler extends FormModalUiHandler {
...
+      if (child instanceof Phaser.GameObjects.Text && child !== this.titleText) {
+        const inputFieldFontSize = languageSettings[i18next.resolvedLanguage!]?.inputFieldFontSize;
+        if (inputFieldFontSize) {
+          child.setFontSize(inputFieldFontSize);
+        }
+      }
+    });
+
+    const warningMessageFontSize = languageSettings[i18next.resolvedLanguage!]?.warningMessageFontSize;
+    const label = addTextObject(this.scene, 10, 87, i18next.t("menu:registrationAgeWarning"), TextStyle.TOOLTIP_CONTENT, { fontSize: warningMessageFontSize ?? "42px" });
```
Comment (f-fsantos) [inline src/ui/registration-form-ui-handler.ts]: Put the default "42px" value when warningMessageFontSize is being assigned please, it's easy to miss that the variable may contain a null value

**Review by f-fsantos — CHANGES_REQUESTED** [review]

**Review by SangaraSorama — APPROVED** [review]

**Review by CodeTappert — APPROVED** [review]

**Review by f-fsantos — APPROVED** [review]

**Pull request merged**